### DISCUSSION TITLE
Generate the part of the Schema programmatically and the rest via @GraphQLApi

### DISCUSSION THREAD

**Discussion by nielsbasjes** [graphql-java/graphql-java#2887] [Q&A] [answered]
Right now I have defined several calls in my API simply by doing something like this:
```
@GraphQLApi
public class Analyzer {

    public static class AnalysisResult {
        public String dummy;
        public AnalysisResult(String foo) {
            dummy = foo + foo;
        }
    }

    @Query("hello")
    public AnalysisResult doHello(@Name("input") String foo) {
        return new AnalysisResult(foo);
    }
}
```
Here the fields in the `AnalysisResult` are found using something like reflection.
This works great for most of my situations except one.

I have a situation where the fields that can be returned by my code should be determined at startup (they can vary depending on system configuration).
So I want the return value to have (from a GraphQL perspective) different fields depending on the configuration at startup (so during a run the list is immutable).
I realize I have to specify both the fields for this part of the Schema and the DataFetchers related to those fields programmatically during startup.

I've been trying to understand the documentation on this subject (i.e. https://www.graphql-java.com/documentation/schema#creating-a-schema-programmatically ) yet these all seem to work in a situation without the use of the annotations like @GraphQLApi. Also the tutorial https://www.graphql-java.com/tutorials/getting-started-with-spring-boot/ works in a radically different way.

I'm looking for a simple Java example that shows me how I can have most code just work with the @GraphQLApi (which obtains the schema through annotations and reflection) and a single API/Query and related parts of the schema that are done programatically.

**Comment by nielsbasjes** [comment, marked as answer]
Via a [different route](https://github.com/spring-projects/spring-graphql/issues/452#issuecomment-1256798212) @bbakerman showed me this which answers my question
https://www.graphql-java.com/documentation/schema/#changing-schema

**Comment by btserket** [comment]
The annotations you're using seem to coming from MicroProfile GraphQL, and not graphql-java? Am I correct?

**Reply by bbakerman** [reply]
> The annotations you're using seem to coming from MicroProfile GraphQL, and not graphql-java? Am I correct?

If I recall right it was Spring annotations so they had to allow Spring Boot create the initial schema and then in a `Customizer` it allowed him to transform the Spring built schema (built via annotations) and edit to it what he needed
